middleware/auth.js

```diff
@@ -83,31 +83,26 @@ module.exports = function (params) {
         try {
           const redirectUri = res.oidc.getRedirectUri();
 
-          let expectedState;
+
           let tokenSet;
           try {
             const callbackParams = client.callbackParams(req);
-            expectedState = transient.getOnce('state', req, res);
-            const max_age = parseInt(
-              transient.getOnce('max_age', req, res),
-              10
-            );
-            const code_verifier = transient.getOnce('code_verifier', req, res);
-            const nonce = transient.getOnce('nonce', req, res);
+            const authVerification = transient.getOnce('auth_verification', req, res);
+
+            const { max_age, code_verifier, nonce, state } = authVerification ? JSON.parse(authVerification) : {} ;
```

Comment:
We could add a check for `authVerification` and throw an error if the cookie doesn't exist.  Right now it currently falls into `BadRequestError: checks.state argument is missing` https://github.com/auth0/express-openid-connect/issues/145 which isn't very helpful for debugging.

Reply:
We can look at that in another story